lib/req.ex

```diff
@@ -762,6 +763,9 @@ defmodule Req do
     * `:receive_timeout` - socket receive timeout in milliseconds, defaults to `15_000`.
 
     * `:unix_socket` - if set, connect through the given UNIX domain socket
+    
+    * `:customize_finch` - a function to modify the built Finch request before execution. This function takes a 
+       Finch request and returns a Finch request. Defaults to the identity function.
```

Comment:
We also need to document this option under `run_finch` :)

Reply:
Yep, I'll finish up the docs and tests now that we like the general approach